scipy/stats/_discrete_distns.py

```diff
@@ -492,18 +492,48 @@ def _entropy(self, M, n, N):
         vals = self.pmf(k, M, n, N)
         return np.sum(entr(vals), axis=0)
 
-    def _sf(self, k, M, n, N):
-        # This for loop is needed because `k` can be an array. If that's the
-        # case, the sf() method makes M, n and N arrays of the same shape. We
-        # therefore unpack all inputs args, so we can do the manual
-        # integration.
+    def _sum_pmf(self, k, M, n, N, cdf):
+        # Sum the tail of the PMF, and then subtract the sum from 1 if
+        # necessary to return the value indicated by the boolean `cdf`.
+        # If k > mode, sum the right tail.  Otherwise sum the left tail.
```

Comment:
I can think of a few reasons you might have chosen the mode as the criterion, but is it always the best choice? What was the reasoning?

Reply:
This and the other comments about code in `_sum_pmf` are now obsolete.  In the update, `_sum_pmf` is gone, and I reimplemented `_cdf` and `_sf` using `_logcdf` and `_logsf`--much simpler!